### PR TITLE
Add autogenerated domain event specs

### PR DESCRIPTION
**Pull request by mxco86**
- Generated domain event message specs from the dev environment

### CODE REVIEW

File: spec/auto/approved-premises/booking-made.yaml
Comment (marcus-bcl) [inline spec/auto/approved-premises/booking-made.yaml:1]: Are we missing booking.changed and booking.cancelled?  Does CAS1 not send those any more?

**Review by marcus-bcl — APPROVED** [review]
Nice!  How are the filenames generated, or are we naming them manually?

File: spec/auto/approved-premises/booking-made.yaml
Comment (mxco86) [reply spec/auto/approved-premises/booking-made.yaml:1]: - The files are named by splitting on the first '.' and then replacing any subsequent '.' chars with '-'. This does highlight where some domain event types are a bit ambiguous

- It only looks for any messages in the test env in the last seven days. I guess we've not raised these ones in the test CAS service. This is a bit of a limitation - there's many more event types raised in prod but we wouldn't want to put the details into the example fields

Comment (marcus-bcl) [reply spec/auto/approved-premises/booking-made.yaml:1]: Ah nice - in that case we can update our `@Message` annotations to just specify the event type, then do that same transformation when we serialize to the AsyncAPI spec.

Good point, I guess there's not much we can do about that - unless we updated our e2e tests to ensure we cover each event type we're interested in?

**Pull request merged**